src/components/icon/icon.ts

```diff
@@ -87,46 +85,62 @@ export default class SlIcon extends ShoelaceElement {
     const library = getIconLibrary(this.library);
     const url = this.getUrl();
 
-    // Create an instance of the DOM parser. We do it here instead of top-level to support SSR while maintaining a
-    // single parser instance for optimal performance.
-    if (!parser) {
-      parser = new DOMParser();
+    if (!url) {
+      this.svg = null;
+      return;
     }
 
-    if (url) {
-      try {
-        const file = await requestIcon(url);
-        if (url !== this.getUrl()) {
-          // If the url has changed while fetching the icon, ignore this request
-          return;
-        } else if (file.ok) {
-          const doc = parser.parseFromString(file.svg, 'text/html');
-          const svgEl = doc.body.querySelector('svg');
-
-          if (svgEl !== null) {
-            svgEl.part.add('svg');
-            library?.mutator?.(svgEl);
-            this.svg = svgEl.outerHTML;
-            this.emit('sl-load');
-          } else {
-            this.svg = '';
-            this.emit('sl-error');
-          }
-        } else {
-          this.svg = '';
-          this.emit('sl-error');
-        }
-      } catch {
+    let iconResolver = iconCache.get(url);
+    if (!iconResolver) {
+      iconResolver = SlIcon._resolveIcon(url);
+      iconCache.set(url, iconResolver);
+    }
+
+    try {
+      const svg = await iconResolver;
+      if (url !== this.getUrl()) {
+        // If the url has changed while fetching the icon, ignore this request
+        return;
+      }
+
+      if (!svg) {
+        this.svg = null;
         this.emit('sl-error');
+        return;
       }
-    } else if (this.svg.length > 0) {
-      // If we can't resolve a URL and an icon was previously set, remove it
-      this.svg = '';
+
+      this.svg = svg.cloneNode(true) as SVGElement;
+      library?.mutator?.(this.svg);
+      this.emit('sl-load');
+    } catch {
+      this.svg = null;
+      iconCache.delete(url);
+      this.emit('sl-error');
     }
   }
 
   render() {
-    return html` ${unsafeSVG(this.svg)} `;
+    return this.svg;
+  }
+
+  private static async _resolveIcon(url: string): Promise<SVGSVGElement | null> {
+    const fileData = await fetch(url, { mode: 'cors' });
+    if (!fileData.ok) return null;
```

Comment:
I don't think we want to be too smart about it. Concerned users can listen for `sl-error` and respond accordingly. I don't believe `<img>` does any retries, but I'm not sure if it caches bad requests. I'd be happy to _not_ cache bad requests, especially if that's how standard images work.

Reply:
I think img handling in browsers deviates a little bit in that it doesn't care about the status code aside from redirects (except maybe 410? I'm not sure about that), but only whether the request body is a valid image or not. For example, if you request a YouTube thumbnail that doesn't exist, it will return a 404 response but still include the default placeholder thumbnail image in the body, and an `img` tag will happily display that.

But from testing, it looks like the browser does not cache failed network requests-- if you add a new img element to the DOM with a URL that was previously seen, but the network request failed to receive a response, it will try again.

Reply:
Got it. Thanks for the info. Sounds like we _shouldn't_ cache bad requests then, but I'm not sure if we want to bake in retries. What were you thinking?

Reply:
We could simply invalidate the cache after a failed request so it tries again next time an icon pops up or needs to be re-rendered. We could cache 410 responses though since they are explicitly cachable.

Reply:
Update the PR to handle retryable errors and cacheable errors